WordPress/gutenberg
# [RNMobile] Link target option on Image Block

## Description
Resolves part of https://github.com/wordpress-mobile/gutenberg-mobile/issues/1337

`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1394

This PR add an extra setting to Image Settings bottom sheet to be able to open links on a new tab (Link target).

This is a simple implementation, where we are **not** adding a new "Links setting bottom sheet" UI.

**Note:** Code-wise, most of the changes are related to trying to share code between web and native. This was all done on https://github.com/WordPress/gutenberg/pull/17675/commits/155633f5c13d933dbafe599f001e53867d07723e, and I can revert it if not needed.

This is the diff without those changes: https://github.com/WordPress/gutenberg/compare/master...204bba9e0a323da1e261f9fa31c4eb5336b61c96

## Screenshots 

![link_target](https://user-images.githubusercontent.com/9772967/65955130-bad76500-e447-11e9-9c5b-d4ae17b6ea37.png)

## To test:
- Build and run the mobile example app.
- On an image block, set a link and turn "Open in a new tab" ON.
- Close the bottom sheet and open the settings again.
- Check that the settings were saved and display as before.
- Press `Clear All Settings`.
- Check that all settings are removed.
